drivers/spi/spi_emul.c

```diff
@@ -117,13 +117,10 @@ static struct spi_driver_api spi_emul_api = {
 	.transceive = spi_emul_io,
 };
 
-#define EMUL_LINK_AND_COMMA(node_id) {		\
-	.label = DT_LABEL(node_id),		\
-},
-
 #define SPI_EMUL_INIT(n) \
-	static const struct emul_link_for_bus emuls_##n[] = { \
-		DT_FOREACH_CHILD(DT_DRV_INST(0), EMUL_LINK_AND_COMMA) \
+	DT_INST_FOREACH_CHILD_SEP(n, EMUL_DT_DECLARE, (;)); \
```

Comment:
I realized I can't do `;` at the end here, it will failt if there are no children, we'll need `_TERM` versions (terminate every entry).